### PR TITLE
Add DEPRECATED Option to add "Deprecated" annotation

### PR DESCRIPTION
**Pull request by ganadist**
Added `DEPRECATED` Option to add `@Deprecated` annotation on each field.
This will help to clean up deprecated FeagureFlag field.

### CODE REVIEW

File: feature-flag/src/main/kotlin/com/linecorp/android/featureflag/FeatureFlagJavaFileWriter.kt [hunk 1]
```diff
@@ -72,14 +72,21 @@ internal class FeatureFlagJavaFileWriter(
         // static analyzer.
         val actualFlagValue = flag.value.toJavaValue(isRequiredLiteralValue)
 
+        if (flag.hasOption(FeatureFlagOption.DEPRECATED)) {
+            emitAnnotation("Deprecated")
```
Comment (munetoshi) [inline feature-flag/src/main/kotlin/com/linecorp/android/featureflag/FeatureFlagJavaFileWriter.kt]: I think `emitAnnotation(Class<? extends Annotation> annotationType)` is a little bit more strict and safer.

File: feature-flag/src/main/kotlin/com/linecorp/android/featureflag/FeatureFlagJavaFileWriter.kt [hunk 2]
```diff
@@ -72,14 +72,21 @@ internal class FeatureFlagJavaFileWriter(
...
+        }
         emitField(JavaType.BOOLEAN, flag.name, flag.getFieldModifier(), actualFlagValue)
+
```
Comment (munetoshi) [inline feature-flag/src/main/kotlin/com/linecorp/android/featureflag/FeatureFlagJavaFileWriter.kt]: `Nit`: Please remove unnecessary blank line.

File: feature-flag/src/main/kotlin/com/linecorp/android/featureflag/FeatureFlagJavaFileWriter.kt [hunk 3]
```diff
@@ -72,14 +72,21 @@ internal class FeatureFlagJavaFileWriter(
...
     }
 
     private fun JavaWriter.writeOverridableFlags(flags: List<FeatureFlagData>) {
         flags.forEach { flag ->
             val flagName = flag.name
             val flagValue = flag.value.toJavaValue(true)
             println("$flagName = $flagValue")
+            if (flag.hasOption(FeatureFlagOption.DEPRECATED)) {
+                emitAnnotation("Deprecated")
```
Comment (munetoshi) [inline feature-flag/src/main/kotlin/com/linecorp/android/featureflag/FeatureFlagJavaFileWriter.kt]: Ditto for line 76.

File: feature-flag/src/main/kotlin/com/linecorp/android/featureflag/model/FeatureFlagOption.kt [hunk 4]
```diff
@@ -38,4 +38,8 @@ internal enum class FeatureFlagOption {
      * This option cannot be used with [OVERRIDABLE] option at the same time.
      */
     LITERALIZE,
+    /**
+     * An option to make a flag value deprecated.
```
Comment (munetoshi) [inline feature-flag/src/main/kotlin/com/linecorp/android/featureflag/model/FeatureFlagOption.kt]: Please rephrase as follows

"""
An option to annotate a flag with \[Deprecated\].
"""

And update line 20 as "...visibility, mutability, and annotations."

File: feature-flag/src/test/kotlin/com/linecorp/android/featureflag/loader/FeatureFlagOptionParserTest.kt [hunk 5]
```diff
@@ -50,7 +51,10 @@ object FeatureFlagOptionParserTest : Spek({
         context("with normal case") {
             val expectedResults = listOf(
                 setOf(PRIVATE),
-                setOf(PRIVATE, OVERRIDABLE)
+                setOf(PRIVATE, OVERRIDABLE),
+                setOf(DEPRECATED),
+                setOf(PRIVATE, DEPRECATED),
+                setOf(PRIVATE, OVERRIDABLE, DEPRECATED)
```
Comment (munetoshi) [inline feature-flag/src/test/kotlin/com/linecorp/android/featureflag/loader/FeatureFlagOptionParserTest.kt:57]: Please add cases for "with duplicated option" as well.

File: feature-flag/src/main/kotlin/com/linecorp/android/featureflag/FeatureFlagJavaFileWriter.kt (hunk 1 above)
Comment (ganadist) [reply feature-flag/src/main/kotlin/com/linecorp/android/featureflag/FeatureFlagJavaFileWriter.kt]: Done

File: feature-flag/src/main/kotlin/com/linecorp/android/featureflag/FeatureFlagJavaFileWriter.kt (hunk 2 above)
Comment (ganadist) [reply feature-flag/src/main/kotlin/com/linecorp/android/featureflag/FeatureFlagJavaFileWriter.kt]: Done

File: feature-flag/src/main/kotlin/com/linecorp/android/featureflag/FeatureFlagJavaFileWriter.kt (hunk 3 above)
Comment (ganadist) [reply feature-flag/src/main/kotlin/com/linecorp/android/featureflag/FeatureFlagJavaFileWriter.kt]: Done

File: feature-flag/src/main/kotlin/com/linecorp/android/featureflag/model/FeatureFlagOption.kt (hunk 4 above)
Comment (ganadist) [reply feature-flag/src/main/kotlin/com/linecorp/android/featureflag/model/FeatureFlagOption.kt]: Done

File: feature-flag/src/test/kotlin/com/linecorp/android/featureflag/loader/FeatureFlagOptionParserTest.kt (hunk 5 above)
Comment (ganadist) [reply feature-flag/src/test/kotlin/com/linecorp/android/featureflag/loader/FeatureFlagOptionParserTest.kt:57]: Done

**Review by munetoshi — APPROVED** [review]
LGTM

**Pull request merged**